client/dashboard/sites/reset-site-modal/index.tsx

```diff
@@ -0,0 +1,242 @@
+import {
+	siteResetContentSummaryQuery,
+	resetSiteMutation,
+	siteResetStatusQuery,
+} from '../../app/queries';
+import ExportNotice from '../export-content-notice';
+import ContentInfo from './content-info';
+import type { Site, SiteContent, SiteResetStatus } from '../../data/types';
+
+import './style.scss';
+
+interface ErrorResponse {
+	code: string;
+	message: string;
+}
+
+function ErrorContent( { message, onClose }: { message: string; onClose: () => void } ) {
+	return (
+		<VStack spacing={ 6 }>
+			<Text>{ message }</Text>
+			<HStack justify="flex-end">
+				<Button variant="primary" onClick={ onClose }>
+					{ __( 'Ok' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+function InProgressContent( { progress }: { progress: number | undefined } ) {
+	const progressValue = progress ? progress * 100 : 10;
+	return (
+		<VStack spacing={ 4 }>
+			<ProgressBar className="reset-site-modal-progress-bar" value={ progressValue } />
+			<Text>{ __( "We're resetting your site. We'll email you once it's ready." ) }</Text>
+		</VStack>
+	);
+}
+
+function ResetSiteContent( {
+	site,
+	siteContent,
+	siteDomain,
+	onSuccess,
+	onError,
+	onClose,
+}: {
+	site: Site;
+	siteContent: SiteContent;
+	siteDomain: string;
+	onSuccess: () => void;
+	onError: ( message: string ) => void;
+	onClose: () => void;
+} ) {
+	const { mutate } = useMutation< void, ErrorResponse >( resetSiteMutation( site.ID ) );
+	const [ isMutatePending, setIsMutatePending ] = useState( false );
+	const [ textValue, setTextValue ] = useState< string >( '' );
+
+	const handleReset = () => {
+		if ( isMutatePending ) {
+			return;
+		}
+
+		setIsMutatePending( true );
+		mutate( undefined, {
+			onSuccess: () => {
+				setIsMutatePending( false );
+				onSuccess();
+			},
+			onError: ( error: ErrorResponse ) => {
+				setIsMutatePending( false );
+
+				if ( error.code === 'staging_sites_not_supported' ) {
+					onError( error.message );
+				} else {
+					onError( __( 'The site could not be reset due to an error. Please try again.' ) );
+				}
+			},
+		} );
+	};
+
+	return (
+		<VStack spacing={ 6 }>
+			<ExportNotice siteSlug={ site.slug } />
+			<Text as="p">
+				{ sprintf(
+					/* translators: %s: site domain */
+					__(
+						"Resetting %s will remove all of its content but keep the site and its URL up and running. You'll also lose any modifications you've made to your current theme. This cannot be undone."
+					),
+					siteDomain
+				) }
+			</Text>
+			<ContentInfo siteContent={ siteContent } siteDomain={ siteDomain } />
+			<TextControl
+				__nextHasNoMarginBottom
+				__next40pxDefaultSize
+				label={ __( 'Type the site domain to confirm' ) }
+				value={ textValue }
+				onChange={ ( value ) => setTextValue( value.trim() ) }
+				help={ sprintf(
+					/* translators: %s: site domain */
+					__( 'The site domain is: %s' ),
+					siteDomain
+				) }
+			/>
+			<HStack spacing={ 4 } justify="flex-end">
+				<Button variant="link" onClick={ onClose }>
```

Comment:
I think this should be `tertiary` so that we don't render underline as per design.

```suggestion
				<Button variant="tertiary" onClick={ onClose }>
```

<img width="197" alt="image" src="https://github.com/user-attachments/assets/7a251a48-fe44-4f71-8813-99c1a54bfcb2" />